### PR TITLE
[tune] fixed validation for search metrics

### PR DESCRIPTION
**Pull request by raoul-khour-ts**
Since advanced search algorithms allow for multiple search metrics we should accommodate for that. Unfortunately, validate result currently only assumes that the search algorithm metric is a string; however, if using sigopt it will be a list. This change was introduced recently and broke some of our stuff, I know that the test is hard to run since you guys might not have the API keys for sigopt... 

https://github.com/ray-project/ray/issues/11581

https://github.com/ray-project/ray/issues/11581

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at

### CODE REVIEW

**Comment by raoul-khour-ts** [comment]
@richardliaw do you mind taking a look at this one. should be kinda straight forward but would be really nice to get out asap since its blocking some stuff on our side (I am reverting back to 1.0 to get around it but then have issues with gcs that i can kinda get around 😐)

**Comment by richardliaw** [comment]
@raoul-khour-ts can you run the linting script: https://docs.ray.io/en/master/getting-involved.html#lint-and-formatting?

**Comment by raoul-khour-ts** [comment]
@richardliaw nice catch, sorry about that, I thought I ran the linter but I guess not. Anyway reuploaded.

**Review by krfricke — CHANGES_REQUESTED** [review]
Can we just report the metrics that are actually missing? E.g. if I expect `["a", "b", "c"]` but only `a` is reported, it should report that `["b", "c"]` are missing. Also it would be great if you could adjust the error string to include the plural form (e.g. "metric(s)".

As a final nit, it would be nice if it doesn't report a list if it is just one metric. The normal case will still be users optimizing only one metric, so we should make sure the outputs look nice in this case.

Thanks!

**Comment by raoul-khour-ts** [comment]
@krfricke good idea :+1: I made the requested changes.

File: python/ray/tune/trial_runner.py [hunk 1]
```diff
@@ -807,16 +807,23 @@ def _validate_result_metrics(self, result):
...
                 location = type(self._scheduler_alg).__name__
-            elif search_metric and search_metric not in result:
-                report_metric = search_metric
+            elif search_metrics and any([
+                    search_metric not in result
+                    for search_metric in search_metrics
+            ]):
+                report_metric = list(
+                    filter(lambda search_metric: search_metric not in result,
+                           search_metrics))
```
Comment (krfricke) [inline python/ray/tune/trial_runner.py:826]:
```suggestion
                           search_metrics))
                   if len(report_metric) == 1:
                       report_metric = report_metric[0]
```

Comment (krfricke) [reply python/ray/tune/trial_runner.py:826]: (please check for correct indentation and linting)

**Comment by krfricke** [comment]
Thanks - once that final nit is addressed I'm happy to approve

**Review by krfricke — APPROVED** [review]

File: python/ray/tune/trial_runner.py (hunk 1 above)
Comment (raoul-khour-ts) [reply python/ray/tune/trial_runner.py:826]: LGTM :+1: I added this (and linted it)

**Comment by richardliaw** [comment]
awesome! just ping after lint/tune tests run.

**Comment by raoul-khour-ts** [comment]
will do

**Comment by krfricke** [comment]
@raoul-khour-ts there seems to be another linting error. Did you run `./ci/travis/format.sh`?

**Comment by raoul-khour-ts** [comment]
@krfricke thanks for the heads up strange I did run it but somehow my linter did not care about the any([generator]) which the ci wanted just any(generator). 🤷‍♂️

**Pull request merged**